### PR TITLE
fix(sum): returns 0 when is empty matching

### LINKED ISSUE #6299: .sum on an empty matching set results in NaN
## What you are doing?

I'm trying to use the SUM aggregate on an empty matching set

``` js
//imagine we have Payments table empty yet
Payment.sum('amount').then(console.log)
//logs NaN
```
## What do you expect to happen?

I'd like the value in this case to be `0` since by definition, a sum is a number, not a NaN.
## Why this happens?

I've quickly found that this happens due to this:

``` js
//node_modules/sequelize/lib/query-interface.js:702
//result contains null as we don't have any matching records
result = parseFloat(result);
```
## What could be done?

since the dataType requested is `dataType instanceof DataTypes.DECIMAL || dataType instanceof DataTypes.FLOAT` as stated in `node_modules/sequelize/lib/query-interface.js:701`, we can additionally check whether the `result` variable  contains `null` and in this case return `0`. I can submit a PR with this fix easily though I'd first like to ensure that I'm not the only one expecting `0` in this case rather than `NaN`.

__Dialect: postgres
__Database version: 9.5.3
__Sequelize version: 3.19.3

### PR DESCRIPTION
**Pull request by t-bonatti**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Fix #6299

### CODE REVIEW

**Comment by t-bonatti** [comment]
@sushantdhiman Could you take a look at this fix too?

**Comment by t-bonatti** [comment]
@sushantdhiman  

I changed to return 0.
I returned `null` before because the Postgres do that.

image

File: lib/query-interface.js [hunk 1]
```diff
@@ -1117,7 +1117,9 @@ class QueryInterface {
         const dataType = options.dataType;
 
         if (dataType instanceof DataTypes.DECIMAL || dataType instanceof DataTypes.FLOAT) {
-          result = parseFloat(result);
+          if (!_.isNull(result)) {
```
Comment (t-bonatti) [inline lib/query-interface.js]: I kept this change to not return `NaN`

Comment (sushantdhiman) [reply lib/query-interface.js]: :+1:

**Comment by sushantdhiman** [comment]
Thanks @t-bonatti 


> I returned null before because the Postgres do that.

Correct behaviour if we want to know if record exists as per our condition, but I can't say other dialect adheres to this. We can simply return numerical values from `aggregate` for now

**Comment by mariomixtega** [comment]
Is this already on 4.42.0?

**Comment by holm** [comment]
I am just upgrading to Sequelize 5, and this caught me by surprise. It seems very strange that this method now returns 0 if the result is null. This seems to assume that the result of an aggregation is a number. This is not at all the case in general. It can really be any datatype, and by returning 0 as a default, this breaks the expected type. In our case we are using it to find the max of a date field.

**Pull request merged**